### PR TITLE
Cleanup tag bundle dependencies

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

Cleanup tag bundle dependencies.

#### Why?

Less dependencies = less circular reference problems in tag manager.

#### BC Breaks/Deprecations

See UPGRADE.md

#### To Do

- [x] Add breaking changes to UPGRADE.md

### CODE REVIEW

File: src/Sulu/Bundle/TagBundle/Resources/config/services.xml [hunk 1]
```diff
@@ -3,43 +3,30 @@
...
             <tag name="sulu.admin"/>
             <tag name="sulu.context" context="admin"/>
             <argument type="service" id="sulu_security.security_checker"/>
         </service>
-        <service id="sulu_tag.tag_repository" class="%sulu_tag.tag_repository.class%">
-            <factory method="getRepository" service="doctrine.orm.entity_manager"/>
-            <argument>%sulu_tag.entity.tag%</argument>
-        </service>
-        <service id="sulu_tag.tag_manager" class="%sulu_tag.tag_manager.class%" public="true">
+        <service id="sulu_tag.tag_manager" class="Sulu\Bundle\TagBundle\Tag\TagManager" public="true">
```
Comment (danrot) [inline src/Sulu/Bundle/TagBundle/Resources/config/services.xml]: Does it need to be public? From my understanding we were only doing this for the services that absolutely need it.

**Comment by danrot** [comment]
Oh, and the tests seem to be failing.

File: src/Sulu/Bundle/TagBundle/Resources/config/services.xml (hunk 1 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/TagBundle/Resources/config/services.xml]: Was public before (little tricky in this diff as I removed the repository).
Currently it needs to be public as it is used in `TagController` and `SmartContentItemController`

File: UPGRADE.md [hunk 2]
```diff
@@ -2,6 +2,17 @@
...
+
+Following Services changed:
+
+ - `sulu_tag.tag_repository` removed use `sulu.repository.tag`
+
+Following Function removed:
+
+ - `TagManagerInterface::getFieldDescriptors` and `TagManagerInterface::getFieldDescriptor` removed use `FieldDescriptorFactory::getFieldDescriptorForClass` instead
+
+The `TagManager` doesn't longer depend on FieldDescriptorFactory, TagEntityName and UserRepository so its constructor changed.
```
Comment (danrot) [inline UPGRADE.md]: Use backticks for the latter 3 classes as well.

File: src/Sulu/Bundle/TagBundle/Tag/TagManager.php [hunk 3]
```diff
@@ -172,16 +119,13 @@ public function save($data, $userId, $id = null)
                 $tag = $this->tagRepository->createNew();
             }
 
-            $user = $this->userRepository->findUserById($userId);
-
             // update data
             $tag->setName($name);
-            $tag->setChanger($user);
```
Comment (danrot) [inline src/Sulu/Bundle/TagBundle/Tag/TagManager.php]: Just a small reminder for me: Test if this still works.

**Pull request merged**